### ISSUE TITLE
Adding support for custom Scalar types

### ISSUE THREAD

**Issue by moufmouf** [thecodingmachine/graphqlite#45]
We should find a way to add support for custom GraphQL scalar types.

Currently support for the DateTime custom scalar type is hard-coded and third-party developers cannot add their own scalar types.

Note: this should be possible using the new `RawTypeMapper` interface described in https://github.com/thecodingmachine/graphqlite/issues/43

Implementation for a custom any-scalar type (accepting string/bool/float...) would go like this:

```php
class ScalarTypeMapper implements RawTypeMapper {
    private function toGraphQLType(Type $type, ?GraphQLType $subType, ReflectionMethod $refMethod, DocBlock $docBlockObj): ?GraphQLType {
        return null;
    }

    private function toGraphQLInputType(Type $type, ?GraphQLType $subType, string $argumentName, ReflectionMethod $refMethod, DocBlock $docBlockObj): ?GraphQLType {
        if ($type instanceof phpDocumentor\Reflection\Type\Scalar) {
            // AnyScalarType is a class implementing the Webonyx ScalarType type.
            return AnyScalarType::getInstance();
        }
    }
}
```

Registering a RawTypeMapper is done at GraphQL boot time:

```php
$schemaFactory->registerRawTypeMapper(new ScalarTypeMapper());
```

**Comment by golaod** [comment]
:+1:

**Comment by moufmouf** [comment]
Ok, this was less easy than it sounded but I finally made it through.

This is fixed by #55 #56 #57.
In the process, I had to bump the major version number because of many breaking changes, so this is only supported in GraphQLite 0.4.

I added a new library that sports a sample GraphQL scalar type: https://github.com/thecodingmachine/graphqlite-misc-types
